simpledotorg/simple-server
# Add profiling RAILS_ENV and make it easy to run Datadog in local dev

**Story card:** [ch4985](https://app.shortcut.com/simpledotorg/story/4985/make-it-easier-to-run-datadog-in-dev-for-local-profiling)

## Because

We want to be able to run Datadog in local dev, and we also want a profiling environment for doing perf / scaling work locally.

## This addresses

Adds a new `profiling` Rails env which is setup to be closer to production - it will cache classes, preload, etc.  It also tackles various tweaks and configuration changes to make sure things play nice when running in this env.

## Test instructions

```
RAILS_ENV=profiling rails server
```

and go hit various pages in dev

You can also see traces from my local dev testing on Datadog [here](https://app.datadoghq.com/apm/traces?query=service%3Asimple_server%20env%3Adevelopment%20operation_name%3Arack.request&cols=core_service%2Ccore_resource_name%2Clog_duration%2Clog_http.method%2Clog_http.status_code&messageDisplay=inline&sort=desc&streamTraces=true&start=1631567216076&end=1631568116076&paused=false).
